src/Orchard/Data/Migration/AutomaticDataMigrations.cs

```diff
@@ -25,30 +29,34 @@ IFeatureManager featureManager
         public ILogger Logger { get; set; } 
 
         public void Activated() {
-
-            // Let's make sure that the basic set of features is enabled.  If there are any that are not enabled, then let's enable them first.
-            var theseFeaturesShouldAlwaysBeActive = new[] {
-                "Common", "Containers", "Contents", "Dashboard", "Feeds", "Navigation", "Scheduling", "Settings", "Shapes", "Title"
-            };
-
-            var enabledFeatures = _featureManager.GetEnabledFeatures().Select(f => f.Id).ToList();
-            var featuresToEnable = theseFeaturesShouldAlwaysBeActive.Where(shouldBeActive => !enabledFeatures.Contains(shouldBeActive)).ToList();
-            if (featuresToEnable.Any()) {
-                _featureManager.EnableFeatures(featuresToEnable, true);
-            }
-
-            foreach (var feature in _dataMigrationManager.GetFeaturesThatNeedUpdate()) {
-                try {
-                    _dataMigrationManager.Update(feature);
-                }
-                catch (Exception e) {
-                    Logger.Error("Could not run migrations automatically on " + feature, e);
+            DistributedLock @lock;
+            if(_distributedLockService.TryAcquireLockForMachine(GetType().FullName, TimeSpan.FromMinutes(30), TimeSpan.FromMilliseconds(250), out @lock)) {

```

Comment:
If multiple threads would potentially access this, then probably so. But I don't know that is the case, since this does not occur from a background task, but in response to the `IOrchardShellEvents.Activated` event. That being said, perhaps it wouldn't hurt to use the thread scoped variant instead.